lib/embedded_ansible/docker_embedded_ansible.rb

```diff
@@ -0,0 +1,200 @@
+      'name'         => memcached_container_name,
+      'ExposedPorts' => {"11211/tcp" => {}}
+    )
+  end
+
+  def run_web_container
+    run_container(
+      awx_web_image_name,
+      'name'         => awx_web_container_name,
+      'Env'          => awx_env,
+      'Hostname'     => "awxweb",
+      'User'         => "root",
+      'ExposedPorts' => {"8052/tcp" => {}},
+      'HostConfig'   => {
+        'Links'        => [rabbitmq_container_name, memcached_container_name],
+        'PortBindings' => {
+          '8052/tcp' => [{ 'HostPort' => AWX_WEB_PORT, 'HostIp' => "0.0.0.0" }]
+        }
+      }
+    )
+  end
+
+  def run_task_container
+    run_container(
+      awx_task_image_name,
+      'name'         => awx_task_container_name,
+      'Env'          => awx_env,
+      'Hostname'     => "awx",
+      'User'         => "root",
+      'ExposedPorts' => {"8052/tcp" => {}},
+      'HostConfig'   => {
+        'Links' => [rabbitmq_container_name,
+                    memcached_container_name,
+                    awx_web_container_name]
+      }
+    )
+  end
+
+  def run_container(image, create_params = {})
+    Docker::Image.create('fromImage' => image)
+    container = Docker::Container.create(create_params.merge('Image' => image))
+    container.start
+  end
+
+  def stop_container(name)
+    Docker::Container.get(name).kill.delete
+  rescue Docker::Error::NotFoundError
+    nil
+  end
+
+  def container_running?(name)
+    Docker::Container.get(name)
+    true
+  rescue Docker::Error::NotFoundError
+    false
+  end
+
+  def awx_env
+    admin_auth    = find_or_create_admin_authentication
+    database_auth = find_or_create_database_authentication
+    rabbitmq_auth = find_or_create_rabbitmq_authentication
+
+    db_host = database_configuration["host"] || docker_bridge_gateway
+    db_host = docker_bridge_gateway if db_host == "localhost"
+
+    [
+      "SECRET_KEY=#{find_or_create_secret_key}",
+      "DATABASE_NAME=awx",
+      "DATABASE_USER=#{database_auth.userid}",
+      "DATABASE_PASSWORD=#{database_auth.password}",
+      "DATABASE_PORT=#{database_configuration["port"] || 5432}",
+      "DATABASE_HOST=#{db_host}",
+      "RABBITMQ_USER=#{rabbitmq_auth.userid}",
+      "RABBITMQ_PASSWORD=#{rabbitmq_auth.password}",
+      "RABBITMQ_HOST=#{rabbitmq_container_name}",
+      "RABBITMQ_PORT=5672",
+      "RABBITMQ_VHOST=awx",
+      "MEMCACHED_HOST=#{memcached_container_name}",
+      "MEMCACHED_PORT=11211",
+      "AWX_ADMIN_USER=#{admin_auth.userid}",
+      "AWX_ADMIN_PASSWORD=#{admin_auth.password}"
+    ]
+  end
+
+  def docker_bridge_gateway
+    br = Docker::Network.get("bridge")
+    br.info["IPAM"]["Config"].first["Gateway"]
+  end
+
+  def container_names
+    [
+      awx_task_container_name,
+      awx_web_container_name,
+      memcached_container_name,
+      rabbitmq_container_name
+    ]
+  end
+
+  def awx_task_container_name
+    "awx_task"
+  end
+
+  def awx_task_image_name
+    "ansible/awx_task:latest"
```

Comment:
Looking for suggestions around how to make this configurable.

Should I go for creating a new key in `Settings`?

Reply:
👍 for Settings